### DISCUSSION TITLE
Unklarheiten Fragestellung Aufgabe 4 -- Teilaufgabe 1 -- Assignment 1

### DISCUSSION THREAD

**Discussion by kaiserog** [irgroup-classrooms/dis08-2021#28] [Q&A]
1. The first task consists of finding twitch emotes (platform-specific emojis). Print an overview of the number of lines per `.csv` file and in total on the occurrence of the terms `kappa` or `trihard`.

- Ist hierbei nach der (generellen) Anzahl der Zeilen pro CSV Dateien gefragt, oder nach den Zeilen in denen die Terme `kappa` or `trihard` vorkommen? 
- Wird für  _'[...] in total on the occurrence of the terms `kappa` or `trihard`'._ nach der absoluten Häufigkeit dieser Terme gefragt, oder soll die Summe für die Zeilen gebildet werden, in denen diese Terme vorkommen?
- Der Aufbau der Emoticons auf Twitch wird hierbei folgendermaßen beschrieben: https://www.twitch.tv/creatorcamp/en/learn-the-basics/emotes/. Es lassen sich in den CSV Dateien auch einige Emoticons finden, die diesem Aufbau folgen. Jedoch gibt es anscheined auch Emoticons, die durch den reinen Term abgebildet werden. Handelt es sich bei kappa und trihard um Emoticons, die durch den reinen Text zu einem Emoticon werden, oder folgen diese auch einem bestimmten Aufbau wie z.B. zu Beginn und Ende des Terms ein Doppelpunkt? Die Frage ergibt sich aus dem Grund, dass ein bestimmter Aufbau eines Emoticons das reine Wort "trihard" oder "kappa" auschließen würde. D.h. manche User schreiben vielleicht trihard oder kappa, printen damit aber kein Emoticon.

Vielen Dank für eine Klarstellung und Beantwortung dieser Fragen.

Beste Grüße,
Konstantin Kaiser

**Comment by HaakFab** [comment]
Hallo,

Sie haben vollkommen Recht, die Formulierung ist etwas verwirrend. Gemeint ist:
Finden Sie zunächst für alle .csv Dateien die Anzahl der Zeilen, in denen mindestens einer der beiden Terme "kappa" oder "trihard" (keine weiteren Zeichen) vorkommt. In einem Weiteren Schritt (Command), lassen Sie sich die Gesamtanzahl aller Vorkommen ausgeben. Diese sollte also die Summe der Counts aus dem ersten Command entsprechen.
Ich habe die Formulierung entsprechend angepasst!
Hinweis: Auch 4.2 wurde leicht angepasst, hier ist das "and in total" weggefallen. 

Ich hoffe, es ist nun deutlicher, was Sie tun sollen. 
VG,
Fabian Haak
